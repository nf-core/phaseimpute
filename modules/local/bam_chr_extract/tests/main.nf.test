nextflow_process {

    name "Test Process BAM_CHR_EXTRACT"
    script "../main.nf"
    process "BAM_CHR_EXTRACT"

    tag "modules"
    tag "modules_local"
    tag "bam_chr_extract"

    test("Extract chr from bam") {
        when {
            process {
                """
                input[0] =  Channel.fromList([
                    [
                        [ id:'test_single_end_bam' ], // meta map
                        file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
                    ],
                    [
                        [id: 'test2_paired_end_sorted_bam'], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_sorted_bam'], checkIfExists: true)
                    ],
                    [
                        [id: 'test_paired_end_sorted_cram'], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(process.out.chr.collect{path(it[1]).readLines()}).match("chr") }
            )
        }
    }
}
