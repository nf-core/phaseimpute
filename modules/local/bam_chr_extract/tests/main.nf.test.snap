{
    "Extract chr from bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2_paired_end_sorted_bam"
                        },
                        "test2_paired_end_sorted_bam.txt:md5,5ae68a67b70976ee95342a7451cb5af1"
                    ],
                    [
                        {
                            "id": "test_paired_end_sorted_cram"
                        },
                        "test_paired_end_sorted_cram.txt:md5,5ae68a67b70976ee95342a7451cb5af1"
                    ],
                    [
                        {
                            "id": "test_single_end_bam"
                        },
                        "test_single_end_bam.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d",
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d",
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d"
                ],
                "chr": [
                    [
                        {
                            "id": "test2_paired_end_sorted_bam"
                        },
                        "test2_paired_end_sorted_bam.txt:md5,5ae68a67b70976ee95342a7451cb5af1"
                    ],
                    [
                        {
                            "id": "test_paired_end_sorted_cram"
                        },
                        "test_paired_end_sorted_cram.txt:md5,5ae68a67b70976ee95342a7451cb5af1"
                    ],
                    [
                        {
                            "id": "test_single_end_bam"
                        },
                        "test_single_end_bam.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d",
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d",
                    "versions.yml:md5,696c749ab73e8b0a4fcce6ff19999e9d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T14:57:17.861877534"
    },
    "chr": {
        "content": [
            [
                [
                    "chr22"
                ],
                [
                    "chr22"
                ],
                [
                    "MT192765.1"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T14:57:44.989492548"
    }
}